vm/src/obj/objint.rs

```diff
@@ -766,6 +770,49 @@ pub fn to_int(vm: &VirtualMachine, obj: &PyObjectRef, mut base: u32) -> PyResult
     )
 }
 
+fn str_to_int(vm: &VirtualMachine, literal: &str, mut base: u32) -> PyResult<BigInt> {
+    let mut buf = literal.chars().filter(|&c| c != '_').collect::<String>();
```

Comment:
cpython has some rules for underscore in integer literal:
there can be only *one* underscore in front of the digit (not prefix)

examples:
```
>>> int('_0x121223', base=16)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 16: '_0x121223'
>>> int('121223_')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '121223_'
>>> int('121__223')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '121__223'
```